core/src/test/java/io/questdb/test/tools/TestUtils.java

```diff
@@ -865,16 +865,8 @@ public static void assertEventually(Runnable assertion, int timeoutSeconds) {
                     throw error;
                 }
             }
-            try {
-                Thread.sleep(nextSleepingTimeMillis);
-                nextSleepingTimeMillis = Math.min(maxSleepingTimeMillis, nextSleepingTimeMillis << 1);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                long elapsedTimeMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
-                throw new AssertionError("Interrupted before timeout. Expected timeout"
-                        + TimeUnit.SECONDS.toMillis(timeoutSeconds) + " ms. Elapsed time: " + elapsedTimeMillis
-                        +" ms. ");
-            }
+            Os.sleep(nextSleepingTimeMillis);
```

Comment:
this is a small behaviour change - Os.sleep() is non-interruptable (and it's eating the interrupt flag) Is this intentional?

Reply:
This is very intentional. Spurious interrupts are not welcome in tests and we used to deal with them as hoc. As in wtf do I do need it this exception. A while back we migrated on using Os.sleep() which observes the delays and doesn’t bother us with unwanted exceptions. We use this throughout the code base. This place was the remaining bastion of Thread.sleep(), which is now destroyed